src/chapters/auth/request-filtering.rst

```diff
@@ -0,0 +1,182 @@
+Filtering Requests
+==================
+
+Our application now allows users to register and log in. However, access to pages on the site is NOT yet restricted in any way. Even if a user isn't logged in, they may view any page on the site. To fully implement authentication, we need to be able to check a user's login status on each request, *before* any controller methods are called.
+
+.. index:: ! request filtering
+
+We can do this by filtering requests. **Request filtering** is a mechanism available in many web frameworks. It allows a programmer to carry out certain actions before any controllers are called.
+
+Request Filters in Spring
+-------------------------
+
+A request filter in Spring can be built by extending the built-in class ``HandlerInterceptorAdapter`` and overriding one of its various methods. This class implements an interface called ``HandlerInterceptor``, which `specifies a few methods <https://docs.spring.io/spring-framework/docs/current/javadoc-api/org/springframework/web/servlet/HandlerInterceptor.html>`_ that are called during the request-handling process:
+
+- ``preHandle`` - called *before* a request is handled by a controller
+- ``postHandle`` - called *after* a request is handled by a controller, but *before* the view is rendered
+- ``afterCompletion`` - called *after* the view is rendered
+
+By extending ``HandlerInterceptorAdapter`` and overriding one or more of these methods, we can write code that is executed at a specific point of *every* request to our application. This is exactly what we need to implement authentication. More specifically, we will override ``preHandle`` and check for a user's login status.
+
+Creating ``AuthenticationFilter``
+---------------------------------
+
+In the top-level package of the app, ``org.launchcode.codingevents``, create a class named ``AuthenticationFilter`` that extends ``HandlerInterceptorAdapter``. 
+
+This class will need to access user data, so autowire a ``UserRepository`` field. We will also need to use the ``AuthenticationController.getUserFromSession`` method, so autowire an ``AuthenticationConroller`` field as well.
+
+.. sourcecode:: java
+   :lineno-start: 19
+
+   public class AuthenticationFilter extends HandlerInterceptorAdapter {
+
+      @Autowired
+      UserRepository userRepository;
+
+      @Autowired
+      AuthenticationController authenticationController;
+
+   }
+
+.. admonition:: Note
+
+   For autowiring to work, a class must be Spring-managed. There are many ways a given class may registered as a Spring-managed class, but any class with ``@Controller`` will be automatically registered. Thus, any controller class may be autowired.
+
+Overriding ``preHandle``
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+Here is our initial ``preHandle`` method, with a detailed breakdown below.
+
+.. sourcecode:: java
+   :lineno-start: 27
+
+   @Override
+   public boolean preHandle(HttpServletRequest request,
```

Comment:
Correct on the docs. However, the `sendRedirect`call we use near the bottom throws IOException, which is a checked exception, so we have either handle or re-throw.

Reply:
I think since we emphasize that the signatures should match exactly and send them to the method they are overriding, we should clarify that its okay that the throw statement does not match

Reply:
Good call.